lib/metro_2/records/base_segment.rb

```diff
@@ -50,5 +50,25 @@ class BaseSegment < Record
     alphanumeric_field :postal_code, 9
     alphanumeric_field :address_indicator, 1
     alphanumeric_field :residence_code, 1
+
+    def joint_segment=(joint_segment)
+      @joint_segment = joint_segment
+    end
+
+    def joint_segment
+      @joint_segment
+    end
+
+    def to_metro2
+      if joint_segment
```

Comment:
Similar to above, this method change should allow for multiple different appendages. 

Also IMO it would be beneficial if `record_descriptor_word` were set here automatically. Thoughts?